visibility-sensor.js

```diff
@@ -93,6 +83,33 @@ module.exports = React.createClass({
     return this.props.containment || window;
   },
 
+  addEventListenerWithDebounce: function (target, event, delay) {
+    if (!this.debounceCheck) {
+      this.debounceCheck = {};
+    }
+
+    var timeout;
+
+    var debounce = function() {
+      var context = this, args = arguments;
+      var later = function() {
+        timeout = null;
+        context.check.apply(context, args);
+      };
+      clearTimeout(timeout);
+      timeout = setTimeout(later, delay || 0);
```

Comment:
@eek, `clearTimeout(this.lastTimeout);` was left behind as a mistake. From my understanding, `this.lastTimeout` was added to track the timeout object so we can cleanup on unmount. But, with the addition of the resize listener, I needed to keep track of multiple timeouts separately to clean up. So I moved it into the "debounce info" object that gets created when `addEventListenerWithDebounce` is called (`timeout` property). However, looking back, I realized I made a mistake of statically referring to the timeout object when creating the debounce info object ... I'll fix that to replace it with a function so we can use closure to access the "last" timeout object that's created